src/helpers/authorship.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  Author,
+  Image,
+  MicroformatProperty,
+  MicroformatRoot,
+  Rels,
+} from "../types";
+
+function getPlainText(values: MicroformatProperty[]): string | null {
+  if (values.length === 0) {
+    return null;
+  }
+
+  const value = values[0] as Author;
+  let plainText: string | null;
+  if (value.value !== undefined && typeof value.value === "string") {
+    plainText = value.value;
+  } else if (typeof value === "string") {
+    plainText = value;
+  } else {
+    plainText = null;
+  }
+
+  return plainText && plainText.trim();
+}
+
+const parseAuthor = (hCard: MicroformatRoot) => {
+  // TODO: Figure out how to stop TypeScript complaining about missing `value`
+  const result: Author = {};
+
+  if (hCard.properties !== undefined) {
+    // Use first (or only) name
+    const names = hCard.properties.name as string[];
+    if (names?.length > 0) {
+      result.name = names[0];
+    }
+
+    // Use first (or only) photo
+    const photos = hCard.properties.photo as Image[];
+    if (photos?.length > 0) {
+      const photo = getPlainText(photos);
+      if (photo) {
+        result.photo = photo;
+      }
+    }
+
+    // Use first (or only) URL
+    const urls = hCard.properties.url as string[];
+    if (urls?.length > 0) {
+      result.url = urls[0];
+    }
+  } else if (hCard) {
+    if (URL.canParse(String(hCard))) {
```

Comment:
I'm not sure we can use `URL.canParse()` - see https://developer.mozilla.org/en-US/docs/Web/API/URL/canParse_static#browser_compatibility

Reply:
Ah, think I confused myself with another Node-only project supporting Node v20. I’ll create a utility function that achieves similar.